### PR TITLE
CI: Use matrix for metal3 images building (2nd try)

### PR DESCRIPTION
**Pull request by mquhuy**
2nd trial for https://github.com/Nordix/metal3-dev-tools/pull/599

### CODE REVIEW

**Review by lentzi90 — DISMISSED** [review]

File: ci/jobs/openstack_image_building.pipeline [hunk 1]
```diff
@@ -65,177 +68,107 @@ pipeline {
...
-              /* Generate base ubuntu image */
-              sh "docker run --rm \
-                ${DOCKER_CMD_ENV}\
-                -v ${CURRENT_DIR}:/data \
-                -v ${METAL3_CI_USER_KEY}:/data/id_ed25519_metal3ci \
-                registry.nordix.org/metal3/image-builder \
-                /data/ci/images/gen_base_ubuntu_image.sh \
-                /data/id_ed25519_metal3ci 1"
+            environment {
+              OS_AUTH_URL=os_auth_urls.${REGION}
```
Comment (mquhuy) [inline ci/jobs/openstack_image_building.pipeline:110]: Here is an instance of where the syntax changed

**Review by furkatgofurov7 — APPROVED** [review]
As agreed, let's try it in a test Jenkins server first, before merging this.

**Review by lentzi90 — APPROVED** [review]
Nice work! This looks great (and I learned some groovy :sweat_smile: )!
/lgtm

**Comment by mquhuy** [comment]
> Nice work! This looks great (and I learned some groovy sweat_smile )! /lgtm

Thanks Lennart. I learned a lot as well xD

**Pull request merged**